### DISCUSSION TITLE
TypeScript official support? and gauge-ts status

### DISCUSSION THREAD

**Discussion by tkitsunai** [getgauge/gauge#2508] [General]
Hello,

I'm a big fan of writing tests with Gauge and have always appreciated using this fantastic tool. Thank you for creating it.
Please forgive the use of machine translation for my message, as English is not my first language.

I enjoy using TypeScript and am looking forward to a future where it's officially supported as a language plugin for Gauge.
I've come across [gauge-ts](https://github.com/BugDiver/gauge-ts) and considered using it. However, I noticed that `gauge-ts` has not been maintained for a while.
(The last release was on June 25, 2022.)

I'm hopeful for the maintenance of `gauge-ts` to resume and would appreciate any information about the current status of gauge-ts and Gauge's support for TypeScript.

Please let me know if there's any way I could support or contribute to this effort. Thank you.

**Comment by chadlwilson** [comment]
Gauge itself has only a little ongoing maintenance/development in general.

However for gauge-ts the situation is more complicated as the plugin was written by one of the original Gauge devs, but is not maintained within the 'getgauge' organisation by other gauge maintainers - so other volunteers can't help with merging PRs, releasing etc, and it doesn't have the same automation set up to make maintenance easier.

Anyway, for gauge-ts you're probably best to ask over on that specific repository?

**Comment by tkitsunai** [comment]
Thank you for your guidance regarding the gauge-ts status.

Indeed, it makes sense for me to inquire directly at the specific repository, and I will do so. 

Should there be no reaction, could you advise on where we might continue discussions concerning Gauge's official TypeScript support? 

Thank you for your time regarding this matter.

**Reply by chadlwilson** [reply]
Discussing here is fine, but you probably should read https://github.com/getgauge/gauge?tab=readme-ov-file#important-note

**Reply by tkitsunai** [reply]
Thanks you for your support.
